packages/xo-web/src/xo-app/settings/cloud-configs/index.js

```diff
@@ -0,0 +1,192 @@
+import _ from 'intl'
+import ActionButton from 'action-button'
+import defined from 'xo-defined'
+import React from 'react'
+import SortedTable from 'sorted-table'
+import { addSubscriptions, generateRandomId } from 'utils'
+import {
+  AvailableTemplateVars,
+  DEFAULT_CLOUD_CONFIG_TEMPLATE,
+} from 'cloud-config'
+import { Container, Col } from 'grid'
+import { find } from 'lodash'
+import { injectState, provideState } from '@julien-f/freactal'
+import { Text } from 'editable'
+import { Textarea as DebounceTextarea } from 'debounce-input-decorator'
+import {
+  createCloudConfig,
+  deleteCloudConfigs,
+  editCloudConfig,
+  subscribeCloudConfigs,
+} from 'xo'
+
+// ===================================================================
+
+const COLUMNS = [
+  {
+    itemRenderer: _ => _.id.slice(4, 8),
+    name: _('formId'),
+    sortCriteria: _ => _.id.slice(4, 8),
+  },
+  {
+    itemRenderer: ({ id, name }) => (
+      <Text value={name} onChange={name => editCloudConfig(id, { name })} />
+    ),
+    sortCriteria: 'name',
+    name: _('formName'),
+    default: true,
+  },
+]
+
+const ACTIONS = [
+  {
+    handler: deleteCloudConfigs,
+    icon: 'delete',
+    individualLabel: _('deleteCloudConfig'),
+    label: _('deleteSelectedCloudConfigs'),
+    level: 'danger',
+  },
+]
+
+const INDIVIDUAL_ACTIONS = [
+  {
+    handler: (cloudConfig, { populateForm }) => populateForm(cloudConfig),
+    icon: 'edit',
+    label: _('editCloudConfig'),
+    level: 'primary',
+  },
+]
+
+const initialParams = {
+  cloudConfigToEditId: undefined,
+  name: '',
+  template: undefined,
+}
+
+export default [
+  addSubscriptions({
+    cloudConfigs: subscribeCloudConfigs,
+  }),
+  provideState({
+    initialState: () => ({
+      formId: generateRandomId(),
+      inputNameId: generateRandomId(),
+      inputTemplateId: generateRandomId(),
```

Comment:
Why do you need generated IDs?

Reply:
To associate the input to its label.

Reply:
Ok but my question is why does it need to be generated and random? Why can't you use `'cloudConfigTemplate'` for example?

Reply:
Because IMHO, it's preferable to avoid static `id`s in components as they are global and it could cause collisions.

Reply:
Ok, then quite a lot of `form`s and `input`s would need to be fixed :)

Reply:
Let's start by not introducing more :wink: